climpred/bootstrap.py

```diff
@@ -234,6 +236,108 @@ def create_pseudo_members(control):
     )
 
 
+def bootstrap_uninit_pm_ensemble_from_control_cftime(init_pm, control):
+    """
+    Create a pseudo-ensemble from control run. Bootstrap random numbers for years to
+    construct an uninitialized ensemble from. This assumes a continous control
+    simulation without gaps.
+
+    Note:
+        Needed for block bootstrapping  a metric in perfect-model framework. Takes
+        random segments of length ``block_length`` from control based on ``dayofyear``
+        (and therefore assumes a constant climate control simulation) and rearranges
+        them into ensemble and member dimensions.
+
+    Args:
+        init_pm (xarray object): ensemble simulation.
+        control (xarray object): control simulation.
+
+    Returns:
+        uninit_pm (xarray object): uninitialized ensemble generated from control run.
+    """
+    check_lead_units_equal_control_time_stride(init_pm, control)
+    # short cut if annual leads
+    if init_pm.lead.attrs['units'] == 'years':
+        return bootstrap_by_stacking(init_pm, control)
+
+    block_length = init_pm.lead.size
+    freq = get_lead_cftime_shift_args(init_pm.lead.attrs['units'], block_length)[1]
+    nmember = init_pm.member.size
+    c_start_year = control.time.min().dt.year.astype('int')
```

Comment:
Doesn't this bake in the assumption that it's annual initializations? What if there are e.g. monthly initializations for a perfect model?

Reply:
no, this only sets the limits for the years to resample from. then with `find_start_date` I subselect the approprimate timestep during the year (to get seasonality right)